lib/image.php

```diff
@@ -260,6 +260,26 @@ function __errorHandler($errno=NULL, $errstr, $errfile=NULL, $errline=NULL, $err
 		$etag = NULL;
 	}
 
+	// Allow CORS
+	// respond to preflights
+	if ($settings['image']['allow_origin'] !== null) {
+		if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
+			// return only the headers and not the content
+			// only allow CORS if we're doing a GET - i.e. no sending for now.
+			if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD']) && $_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD'] == 'GET') {
+				header('Access-Control-Allow-Origin: *');
```

Comment:
Good catch. Not sure if needed on preflight (not sure as in “don’t know”…) But sure it sounds logical to be consistent.